### PR TITLE
Added migration script for new mobiledoc_revisions table

### PR DESCRIPTION
**Pull request by rshbhgrg**
refs https://github.com/TryGhost/Ghost/pull/9929

- New migration script for `mobiledoc_revisions` table which will be used to store multiple mobiledoc versions

### CODE REVIEW

**Comment by rishabhgrg** [comment]
@gargol I was actually not sure if we need to include changes in schema or model as part of this PR or if that is separate. Good catch 🎣 Updated

File: core/server/data/migrations/versions/2.2/2-add-mobiledoc-revisions-table.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const common = require('../../../../lib/common');
```
[the remark this answers is not in the document]
Comment (rishabhgrg) [reply core/server/data/migrations/versions/2.2/2-add-mobiledoc-revisions-table.js]: Updated

File: core/server/data/schema/schema.js [hunk 2]
```diff
@@ -308,5 +308,12 @@ module.exports = {
         session_data: {type: 'string', maxlength: 2000, nullable: false},
         created_at: {type: 'dateTime', nullable: false},
         updated_at: {type: 'dateTime', nullable: true}
+    },
+    mobiledoc_revisions: {
+        id: {type: 'string', maxlength: 24, nullable: false, primary: true},
+        post_id: {type: 'string', maxlength: 24, nullable: false, index: true},
```
Comment (kirrg001) [inline core/server/data/schema/schema.js]: > index: true

Did you double check that? Looks like the wrapper does not support index currently

Comment (gargol) [reply core/server/data/schema/schema.js]: confirming it's not working, the schema in mysql looks like this after migration:
```
| mobiledoc_revisions | CREATE TABLE `mobiledoc_revisions` (
  `id` varchar(24) NOT NULL,
  `post_id` varchar(24) NOT NULL,
  `mobiledoc` longtext,
  `created_at_ts` bigint(20) NOT NULL,
  `created_at` datetime NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
```

Comment (kirrg001) [reply core/server/data/schema/schema.js]: @rishabhgrg You just removed it? Why is that?

Comment (rishabhgrg) [reply core/server/data/schema/schema.js]: @kirrg001 We just tested it with @gargol, it does not supports index right now so removed it quickly. We were just discussing if we should add index support as part of these changes or we leave it for now

Comment (gargol) [reply core/server/data/schema/schema.js]: @kirrg001 based on https://github.com/TryGhost/Ghost/blob/master/core/server/data/schema/commands.js#L8-L46, we don't seem to support indexes at the moment right?

Comment (kirrg001) [reply core/server/data/schema/schema.js]: Yeah based on my assumption and based on the confirmation [comment](https://github.com/TryGhost/Ghost/pull/9934#discussion_r221980335) from @gargol, i noticed it's not supported.

@rishabhgrg 
Just removing "index: true", force push the commit and not leaving a comment why you decided that way, is a little invisible. The discussion was even hided after force push, which makes is hart to keep track of what's going on.

I'd suggest to double check knex docs and see what's missing to support the index?

Comment (rishabhgrg) [reply core/server/data/schema/schema.js]: @kirrg001 Hmm yeah sorry was bit too over-reactive and pushing changes too fast to push this for today's release keeping a single commit, its not helpful 😅Will look into knex docs for index feature, should I meanwhile revert anything for cleaner history or leave as is for now?

Comment (kirrg001) [reply core/server/data/schema/schema.js]: >  should I meanwhile revert anything for cleaner history or leave as is for now?

no history looks fine 👍

Comment (kirrg001) [reply core/server/data/schema/schema.js]: @rishabhgrg do you need help?

Comment (rishabhgrg) [reply core/server/data/schema/schema.js]: Just finished updating, `index` is in place in wrapper using knex command, tested on local mysql and it seems to work
screen shot 2018-10-02 at 9 48 28 pm

File: core/server/data/schema/commands.js [hunk 3]
```diff
@@ -43,6 +43,9 @@ function addTableColumn(tableName, table, columnName) {
     if (columnSpec.hasOwnProperty('defaultTo')) {
         column.defaultTo(columnSpec.defaultTo);
     }
+    if (columnSpec.hasOwnProperty('index') && columnSpec.index === true) {
+        column.index();
+    }
```
Comment (rishabhgrg) [inline core/server/data/schema/commands.js]: @kirrg001 Added `column.index` property from knex from this doc - https://knexjs.org/#Schema-alter

File: core/server/data/schema/schema.js (hunk 2 above)
Comment (kirrg001) [reply core/server/data/schema/schema.js]: Cool. Where is this handled in our command wrapper? Can you pls show me the code section? Thanks 👍

File: core/server/data/schema/commands.js (hunk 3 above)
Comment (kirrg001) [reply core/server/data/schema/commands.js]: Ah okay, i didn't see the extra commit 👍

Comment (kirrg001) [reply core/server/data/schema/commands.js]: Could you now please make two clean commits?

1. commands support for index
2. the actual commit

Thanks 👍

Comment (kirrg001) [reply core/server/data/schema/commands.js]: Then i can rebase the PR

File: core/server/data/schema/schema.js (hunk 2 above)
Comment (kirrg001) [reply core/server/data/schema/schema.js]: 💃

**Comment by kirrg001** [comment]
@rishabhgrg This branch has conflicts that must be resolved

👯‍♂️

**Review by kirrg001 — APPROVED** [review]

**Pull request merged**